app/assets/javascripts/discourse/app/routes/build-category-route.js

```diff
@@ -18,64 +13,62 @@ import {
 import DiscourseRoute from "discourse/routes/discourse";
 import I18n from "discourse-i18n";
 
+@disableImplicitInjections
 class AbstractCategoryRoute extends DiscourseRoute {
   @service composer;
   @service router;
+  @service store;
+  @service topicTrackingState;
+  @service("search") searchService;
 
   queryParams = queryParams;
 
-  model(modelParams) {
+  templateName = "discovery/list";
+  controllerName = "discovery/list";
+
+  async model(params, transition) {
     const category = Category.findBySlugPathWithID(
-      modelParams.category_slug_path_with_id
+      params.category_slug_path_with_id
     );
 
     if (!category) {
-      const parts = modelParams.category_slug_path_with_id.split("/");
-      if (parts.length > 0 && parts[parts.length - 1].match(/^\d+$/)) {
-        parts.pop();
-      }
-
-      return Category.reloadBySlugPath(parts.join("/")).then((result) => {
```

Comment:
This recovery logic was introduced to handle category slug changes. Nowadays, our category URLs include a numeric ID in the title so this is no longer a concern.